python/cudf/cudf/core/series.py

```diff
@@ -3658,6 +3659,17 @@ def where(self, cond, other=None, inplace=False):
             inplace=inplace,
         )
 
+    @_cudf_nvtx_annotate
+    @_warn_no_dask_cudf
+    def __dask_tokenize__(self):
+        from dask.base import normalize_token
+
+        return [
+            type(self),
+            str(self.dtype),
+            normalize_token(self.to_pandas()),
+        ]
```

Comment:
I'm honestly not sure why the `Frame.__dask_tokenize__` definition (which looks very similar) isn't being used?

Whatever the reason may be, `test_categorical_compare_ordered` fails without this fix, because **different** `Series` objects end up being tokenized to the same value, and the corresponding expressions are cached between tests. (**general message**: Unique/proper tokenization is very important when query-planning is active)

Reply:
Looks like before this was using `IndexedFrame.__dask_tokenize__` where the data was being hashed as `self.hash_values().values_host` as opposed to `self.to_pandas()` (might be difference here for categorical)

Might be a better fix if `IndexedFrame.__dask_tokenize__` uses `to_pandas()` instead of `self.hash_values`? Additionally if we want to use this fix as-is I think we would need to also incorporate the `self.index`?

Reply:
>Looks like before this was using IndexedFrame.__dask_tokenize__ where the data was being hashed as self.hash_values().values_host as opposed to self.to_pandas() (might be difference here for categorical)

Aha! Thanks for pointing that out @mroeschke !

This is actually a problem we ran into before and fixed in `Frame.__dask_tokenize__`. It turns out that `normalize_token(self._dtypes)` doesn't work very well. The more reliable thing to do is actually use `str(self._dtypes)`. With that said, dtypes with many categories may not be completely/well represented by `str(self._dtypes)`. Therefore, I just added an extra line to explicitly normalize the actual `categories` for each categorical dtype.

>Might be a better fix if IndexedFrame.__dask_tokenize__ uses to_pandas() instead of self.hash_values?

I think you are right that this is probably the safest and most robust thing to do. However, I am still hesitant to remove the `hash_values` code path. Right now, we avoid moving more than two columns (the hashed values, and the index) to host memory when a cudf object is tokenized. The overhead difference may not be dramatic, but it would be nice to avoid moving the whole thing to pandas.